crates/swc_compiler_base/src/lib.rs

```diff
@@ -37,17 +37,28 @@ pub struct TransformOutput {
 
     #[serde(skip_serializing_if = "Option::is_none")]
     pub output: Option<String>,
+
+    pub diagnostics: std::vec::Vec<String>,
 }
 
 #[cfg(not(feature = "node"))]
 #[derive(Debug, Serialize)]
 pub struct TransformOutput {
     pub code: String,
+
     #[serde(skip_serializing_if = "Option::is_none")]
     pub map: Option<String>,
 
     #[serde(skip_serializing_if = "Option::is_none")]
     pub output: Option<String>,
+
+    pub diagnostics: std::vec::Vec<String>,
+}
+
+impl DiagnosticEmitter for TransformOutput {
+    fn emit(&mut self, diagnostic: String) {
+        self.diagnostics.push(diagnostic);
```

Comment:
I think this is inefficient without a good reason.

Reply:
Maybe  transformOutput take diagnostics handler directly is a better way.
```rust 
transformOutput.diagnostics = handler.take_diagnostics();
```